File: TGME/src/main/java/tmge/core/GameRegistry.java
package tmge.core;

import tmge.model.Game;
import tmge.model.GameInfo;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class GameRegistry {
    private Map<String, GameFactory> availableGames;

    public GameRegistry() {
        availableGames = new HashMap<>();
        // Register default games
        // This would be where you register your specific game implementations
    }

    public List<GameFactory> getAvailableGames() {
        return new ArrayList<>(availableGames.values());
    }

    public void registerGame(GameFactory factory) {
        GameInfo info = factory.getGameInfo();
        availableGames.put(info.getName(), factory);
    }

    public Game createGame(String gameType) {
        GameFactory factory = availableGames.get(gameType);
        if (factory != null) {
            return factory.createGame();
        }
        return null;
    }
}

File: TMGE_p/registry/game_registry.py
from typing import List, Dict
from game_factory import GameFactory
from game_info import GameInfo

class GameRegistry:
    def __init__(self):
        self.available_games = []
    
    def register_game(self, game_factory: GameFactory) -> None:
        self.available_games.append(game_factory)
    
    def get_available_games(self) -> List[GameInfo]:
        return [factory.get_game_info() for factory in self.available_games]
    
    def create_game(self, game_type: str, players):
        for factory in self.available_games:
            if factory.get_game_info().get_name() == game_type:
                return factory.create_game(players)
        return None